src/App.tsx

```diff
@@ -1,39 +1,63 @@
-import React from 'react';
+import { Component } from 'react';
 import './App.scss';
+import { Clock } from './Components/Clock';
 
 function getRandomName(): string {
   const value = Math.random().toString().slice(2, 6);
 
   return `Clock-${value}`;
 }
 
-export const App: React.FC = () => {
-  const date = new Date();
-  const clockName = getRandomName();
-
-  // This code starts a timer
-  const timerId = window.setInterval(() => {
-    // ...
-  }, 1000);
-
-  // this code stops the timer
-  window.clearInterval(timerId);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-
-      <div className="Clock">
-        <strong className="Clock__name">
-          {clockName}
-        </strong>
-
-        {' time is '}
+export type State = {
+  hasClock: boolean
+  clockName: string,
+};
 
-        <span className="Clock__time">
-          {date.toLocaleTimeString()}
-        </span>
+class App extends Component<{}, State> {
+  state: Readonly<State> = {
```

Comment:
Redundant explicit typing
```suggestion
  state = {
```